src/utils.js

```diff
@@ -13,8 +13,17 @@ export const getComponentVariant = (theme, componentName, variant) => {
 
 export const getComponentStyle = componentName => themeGet(`components.${componentName}.style`);
 
-export const getComponentClassName = ({ className, theme: { classPrefix }, variant }, name) =>
-  `${className || ''} ${classPrefix}-${name} ${variant ? `${classPrefix}-${name}-${variant}` : ''}`.trim();
+const getComponentClassName = flow(
+  ({ className, passedClassName, theme: { classPrefix }, variant }, name) =>
+    `${className || ''} ${passedClassName || ''} ${classPrefix}-${name} ${
```

Comment:
Looks like styled-components automatically applies the className passed on render undo the hood. We can fix the class duplication by removing `className` here.

Could we also change `passedClassName` to `baseClassName` to keep consistent with `baseProps`? :)